CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+
+### Che-theia init
+
+You can use the che command `init ... DEV che-theia` (command pallette > Run task > … or containers view)
+Basically, this command will perform:
+```
+[che-dev]
+$ che:theia init --dev
+```
+
+This command will checkout all extensions in `/projects/theia/che` folder and create symbolic link into `/projects/theia/packages` folder.
+
+By default, extensions list is retrieved from https://github.com/eclipse/che-theia/blob/master/che-theia-init-sources.yml
+
+### Compile Che-Theia
+You can use the che command `build ... DEV che-theia` (command pallette > Run task > … or containers view)
+Basically, this command will perform:
+```
+[che-dev]
+$ yarn
+$ che:theia production
+```
+
+- `yarn` will compile theia + additional Che-theia extensions that have been setted up by `che:theia init`
+- `che:theia production` will generate a production like package.
+
+### Compile your plugin
+You can use the che command `build ... containers-plugin` or `build ... factory-plugin` (command pallette > Run task > … or containers view)
+Basically, this command will perform in the right plugin folder:
+
+```
+[che-dev]
+$ yarn
+```
+
+`yarn` will compile the plugin and produce a .theia file which is the plugin itself.
+
+### Run che-theia + plugin
+You can use the che command `run ... DEV che-theia + factory-plugin` or `run ... DEV che-theia + containers-plugin` (command pallette > Run task > … or containers view)
+Basically, this command will start the DEV che-theia with the plugin:
+
+```
+[theia-ide]
+$ export HOSTED_PLUGIN=/projects/che-theia/plugins/factory-plugin/
+$ node /projects/theia/production/src-gen/backend/main.js /tmp/theiadev_projects --hostname=0.0.0.0 --port=3130
+```
+
+You can then access to your modified Che-theia from the Container view `theia-dev` endpoint
+![Che-Theia-dev-endpoint](https://raw.githubusercontent.com/eclipse/che-theia/assets/theia-dev-endpoint.png)
+
+### Run che-theia + plugin in dev mode
+You can use the che command `run ... DEV yarn start ... che-theia + factory-plugin` or
+`run ... DEV yarn start... che-theia + containers-plugin`
+(command pallette > Run task > … or containers view).
+To start che-theia in `dev-mode` with yarn (not using the production che-theia generated). It would keep things like the ability to source map.
+
+Running dev theia would be located in the che-dev container and `theia-dev-flow` endpoint:
+![Che-Theia-dev-endpoint](https://raw.githubusercontent.com/eclipse/che-theia/assets/theia-dev-flow-endpoint.png)
+
+
+
+### Just want to build the plugin and run with the existing Theia image
+If you do not have any changes on Theia or Che-theia extension, you could just build the plugins with `build ... containers-plugin` or `build ... factory-plugin`
+and run these plugins with the existing che-theia app:
+`run ... HOSTED che-theia + container-plugin` or `run ... HOSTED che-theia + factory-plugin`
+
```

Comment:
Maybe we should also have one more section on about packaging/publishing a plugin. pointing to another doc is fine.